lib/parseSty.js

```diff
@@ -0,0 +1,133 @@
+"use strict";
+
+var balanced = require("balanced-match");
+
+var declarationCommands = ["\\newcommand",
+                           "\\renewcommand",
+                           "\\DeclareRobustCommand",
+                           "\\defSpecFun"];
+
+// Returns a two dimensional array, where each element contains the body of a bracket section and the type of bracket
+// that surrounds it. Makes use of the npm balanced-match module.
+// E.g.: [ [ "in brackets", "[" ], [ "in braces", "{" ] ]
+var findBracketSections = module.exports.findBracketSections = function(string, sections) {
+    if (!sections) {
+        sections = [];
+    }
+    var bracket = balanced("[", "]", string);
+    var brace = balanced("{", "}", string);
+    if (brace && (!bracket || brace.start < bracket.start)) {
+        sections.push([brace.body, "{"]);
+        return findBracketSections(brace.post, sections);
+    }
+    else if (bracket) {
+        sections.push([bracket.body, "["]);
+        return findBracketSections(bracket.post, sections);
+    }
+    return sections;
+};
+
+// Finds the first occurrence in the provided string of the earliest occurring item in the given array, starting from
+// a specified index. Returns both the index of occurrence and the relevant item of subs (returns -1 and the first item
+// if none of the provided substrings can be found).
+function findFirst(string, subs, start) {
+    var min = -1;
+    var index;
+    var first = subs[0];
+    for (var i = 0; i < subs.length; i++) {
+        index = string.indexOf(subs[i], start);
+        if ((index !== -1) && (index < min || min === -1)) {
+            min = index;
+            first = subs[i];
+        }
+    }
+    return [min, first];
+}
+
+// Parses \newcommand .. ..
+function parseNewCommand(definition, command, cmds) {
+    definition = definition.substring(command.length);
+    var sections = findBracketSections(definition);
+    switch (sections.length) {
+        // \newcommand\command1\command2
+        case 0:
+            var commands = definition.match(/\\[a-zA-Z]+/g);
```

Comment:
The pattern originates from the restriction of latexmacronames. Can you add a reference to this fact here?